src/renderer.rs

```diff
@@ -1,211 +0,0 @@
-
-        Ok(())
-    }
-
-    fn draw_green_square(&mut self, point: Point, alpha: u8) -> Result<(), String> {
-        self.draw_square(point, Color::RGBA(107, 111, 70, alpha))?;
-        Ok(())
-    }
-
-    fn draw_square(&mut self, (x, y): Point, color: Color) -> Result<(), String> {
-        let x = x * TILE_SIZE;
-        let y = y * TILE_SIZE;
-
-        self.canvas.set_draw_color(color);
-        self.canvas
-            .fill_rect(Rect::new(x, y, TILE_SIZE as u32, TILE_SIZE as u32))?;
-        Ok(())
-    }
-
-    fn draw_king_threatened(&mut self, (y, x): Point) -> Result<(), String> {
-        let x = x * TILE_SIZE;
-        let y = y * TILE_SIZE;
-
-        let circle_x = x + TILE_SIZE / 2;
-        let circle_y = y + TILE_SIZE / 2;
-        let circle_rad = TILE_SIZE as f64 / 2.0;
-
-        self.draw_blury_circle((x, y), (circle_x, circle_y), circle_rad)?;
-        Ok(())
-    }
-
-    fn draw_blury_circle(
-        &mut self,
-        (x, y): Point,
-        (circle_x, circle_y): Point,
-        circle_rad: f64,
-    ) -> Result<(), String> {
-        for i in x..=(x + TILE_SIZE) {
-            for j in y..=(y + TILE_SIZE) {
-                let distance = (((i - circle_x).pow(2) + (j - circle_y).pow(2)) as f64).sqrt();
-                if distance < circle_rad {
-                    let alpha = 255.0 * (1.0 - (distance / circle_rad));
-                    let alpha = alpha as u8;
-                    self.canvas.set_draw_color(Color::RGBA(255, 0, 0, alpha));
-                    self.canvas.draw_point(rect::Point::new(i, j))?;
-                }
-            }
-        }
-
-        Ok(())
-    }
-
-    fn draw_occupied_move_tile(&mut self, (x, y): (i32, i32)) -> Result<(), String> {
-        let x = x * TILE_SIZE;
-        let y = y * TILE_SIZE;
-        let color = Color::RGBA(110, 110, 70, (u8::MAX as f64 * TRANSPARENCY) as u8);
-
-        let circle_x = x + TILE_SIZE / 2;
-        let circle_y = y + TILE_SIZE / 2;
-        let circle_rad = TILE_SIZE as f64 / 1.75;
-        self.canvas.set_draw_color(color);
-        for i in x..=(x + TILE_SIZE) {
-            for j in y..=(y + TILE_SIZE) {
-                if (((i - circle_x).pow(2) + (j - circle_y).pow(2)) as f64).sqrt() > circle_rad {
-                    self.canvas.draw_point(rect::Point::new(i, j))?;
-                }
-            }
-        }
-
-        Ok(())
-    }
-
-    fn draw_empty_move_tile(&mut self, (x, y): (i32, i32)) -> Result<(), String> {
-        let x = x * TILE_SIZE;
-        let y = y * TILE_SIZE;
-
-        let circle_x = x + TILE_SIZE / 2;
-        let circle_y = y + TILE_SIZE / 2;
-        let circle_rad = TILE_SIZE / 8;
-
-        self.canvas.filled_circle(
-            circle_x as i16,
-            circle_y as i16,
-            circle_rad as i16,
-            Color::RGBA(110, 110, 70, (u8::MAX as f64 * TRANSPARENCY) as u8),
-        )?;
-        Ok(())
-    }
-
-    fn draw_tile(&mut self, (x, y): (i32, i32), tile: Tile) -> Result<(), String> {
```

Comment:
use a struct rather than tuples for parameters

Reply:
Even though it's mostly clear what you mean, it's still easier to read when you use a struct